client/src/components/container/portal/Attendance/index.js

```diff
@@ -0,0 +1,267 @@
+import React from "react";
```

Comment:
Try to break this into smaller components.
Personally this is how I would do it, but there are many ways:

- Everything from line 193 - 259 could go into a separate component, but within the Attendance module, maybe in `components/container/portal/Attendance/AttendanceGrid.js` -> I would nest it here because it's something that will only be used by the Attendance parent component for now
- Make that a simple functional component that takes `events` and the callbacks you need for the form as props.
- Work my way up and try to create more display components from the code here.

If you get to the situation where you have a parent component that loads and formats data, then passes it to child components for display, this is a really nice position to be in, it makes your code easy to test and reason about